aspnetcore/security/authorization/secure-data/samples/final15/Authorization/ContactRoleAuthorizationHandler.cs

```diff
@@ -16,7 +16,7 @@ public class ContactRoleAuthorizationHandler
                 return Task.FromResult(0);
             }
 
-            if (context.User.IsInRole(requirement.Name))
+            if (context.User.IsInRole(Constants.OperationRolePrefix + requirement.Name))
```

Comment:
Ah maybe the sample is too simple then. If you think about it an Administrator is probably able to do anything, so individual permissions end up as meaningless. I'd imagine 3 users, an admin, a user, and a guest. 

Then we'd have permissions of Create, View, Edit, Delete. 

Then for roles we'd have Administrator, and User.

An member of the admin role would be able to do anything. Someone in the user role would be able to create, view, edit their own contacts, and delete their own contacts, A guest, who has no roles, might only have View.

So roles don't map to a single operation/permission, they can map to multiple operations. Does that make sense?

(I know this is going to complicate the sample somewhat, sorry)